ADR/ADR-002-Optional-Dependency-CLI11.md

```diff
@@ -0,0 +1,270 @@
+Moving forward we can either extend the existing implementation or take on a
+third-party dependency.
+
+## Options Considered
+
+We considered build ourselves or take on one of several existing open source
+libraries.
+
+### Extend EcKit
+
+Design and architecture of a command line parsing is sometimes considered a
+'entry level' topic, it nevertheless requires consensus building, design,
+implementation and testing work. Our development time and resources are better
+spent on other topics.
+
+### Open Source Solution
+
+Looking at the available solutions the following characteristics have been
+taken into account:
+
+- Feature richness. -> To ensure the library will serve us well even with
+  changing requirements.
+- Maturity. -> To ensure that the API is stable enough to build on it.
+- Development activity and age. -> We want to pick something that is going to
+  be maintained for a long time. Expected future lifetime of a project is
+  proportional to its current age (Lindy's law).
+- Compatible License -> No GPL
+- Ease of integration 
+
+The considered libraries are:
+
+#### Argprse
+
+[argparse](https://github.com/p-ranav/argparse)
+
+Replicates python argparse in C++.
+
+Relative few commits in the last year, development speed seems to have slowed
+down.
+
+Development started in 2019, 733 commits to date.
+
+Developed under MIT-License.
+
+Distributed as a single header library.
+
+#### CLI11
+
+[CLI11](https://github.com/CLIUtils/CLI11)
+
+Very close to rusts clap library. Most comprehensive list of features of all
+libraries in this list. Allows arbitrary nesting of sub-commands. Out of the
+box support to map configuration files / environment variables to command line
+options.
+
+Active development. Multiple feature / bug fix releases in this year.
+
+Development started in 2017, 1357 commits to date.
+
+Developed BSD Clause 3 License.
+
+Distributed as a single header library.
+
+#### Lyra
+
+[Lyra](https://github.com/bfgroup/Lyra)
+
+Fork of Clara, the command line parsing library used by Catch2. DSL-Style
+interface. Most features can be realised but dos not feature mutually exclusive
+argument groups for example.  
+
+Active but slow development.
+
+Forked in 2019, 513 commits to date.
+
+Developed under Boost Software License v1.
+
+Distributed as a header only library.
+
+#### CmdLime
+
+[cmdlime](https://github.com/kamchatka-volcano/cmdlime)
+
+Declarative style. No support for mutually exclusive groups.
+
+Very little active development.
+
+Development started in 2021, 150 commits to date.
+
+Developed under Microsoft Public License.
+
+Distributed as a header only library.
+
+#### Boost Program Options
+
+[Boost Program Options](https://www.boost.org/doc/libs/1_89_0/doc/html/program_options.html)
+
+Most mature library of the selection but does not offer direct support for
+subcommands, although this can be made to work. It comes with boost. Only
+library of this selection that is not header only.
```

Comment:
I would add that boost builds are slow and heavy and as a dependency historically has given occasionally broken builds / deps due to API changes.

Reply:
I put even more emphasis on it
Added:
>  In general Boost is a cumbersome dependency that historically had issues with compatibility. We do not want to depend on Boost unless we have to.